daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -423,6 +437,16 @@ private[lf] case class PartialTransaction(
         noteAbort(Tx.EndExerciseInRootContext)
     }
 
+  /** Note that the transaction building failed due to an authorization failure */
+  def noteAuthFails(nid: NodeId, fails: List[FailedAuthorization]): PartialTransaction = {
+    fails match {
+      case Nil => this
+      case fa1 :: _ => // take just the first failure
+        val fas = Map(nid -> fa1)
```

Comment:
There's potential for simplification in a future PR here:
1. If we only care about the first error, let's only compute the first error.
2. If the map always contains one entry, let's use a pair instead.

Reply:
again, I'll do this in the following PR